tests/phpunit/integration/Core/Modules/REST_Modules_ControllerTest.php

```diff
@@ -0,0 +1,521 @@
+<?php
+/**
+ * ModulesTest
+ *
+ * @package   Google\Site_Kit
+ * @copyright 2021 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Tests\Core\Modules;
+
+use Google\Site_Kit\Context;
+use Google\Site_Kit\Core\Modules\Modules;
+use Google\Site_Kit\Core\REST_API\REST_Routes;
+use Google\Site_Kit\Core\Storage\Options;
+use Google\Site_Kit\Core\Storage\User_Options;
+use Google\Site_Kit\Tests\TestCase;
+use Google\Site_Kit\Tests\FakeHttpClient;
+use WP_REST_Request;
+
+class REST_Modules_ControllerTest extends TestCase {
+
+	public function set_up() {
+		parent::set_up();
+
+		$this->user         = $this->factory()->user->create_and_get( array( 'role' => 'administrator' ) );
+		$this->context      = new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE );
+		$this->options      = new Options( $this->context );
+		$this->user_options = new User_Options( $this->context, $this->user->ID );
+		$this->modules      = new Modules( $this->context, $this->options, $this->user_options );
+		wp_set_current_user( $this->user->ID );
+
+		// This ensures the REST server is initialized fresh for each test using it.
+		unset( $GLOBALS['wp_rest_server'] );
+		remove_all_filters( 'googlesitekit_rest_routes' );
+
+		$this->modules->register();
+	}
+
+	public function test_register() {
+		remove_all_filters( 'googlesitekit_apifetch_preload_paths' );
+		$this->modules->register();
+
+		$this->assertTrue( has_filter( 'googlesitekit_apifetch_preload_paths' ) );
+		$this->assertTrue( has_filter( 'googlesitekit_features_request_data' ) );
+		$this->assertContains(
+			'/' . REST_Routes::REST_ROOT . '/core/modules/data/list',
+			apply_filters( 'googlesitekit_apifetch_preload_paths', array() )
+		);
+	}
+
+	public function test_check_access_rest_endpoint__no_get_method() {
```

Comment:
While we're at it, let's reorder the tests here so that the methods are tested in the same order they're defined in the source. This will make them a bit easier to look up later.

Reply:
Tests reordered!